File: Stack/C/stack.c
#include <stdio.h>
#include <math.h>
#include <stdlib.h>
#include <string.h>

#define MAX_STACK_SIZE 1000

typedef struct stack
{
    int current_size;
    int items[MAX_STACK_SIZE];//We store the items in the stack using an array, but we can also create stacks with lists
}STACK;

//This function initializes our stack
STACK* create_stack();

//This function inserts a new item on stack
void push(STACK *Stack, int item);

//This function removes an item from stack
int pop(STACK *Stack);

//This function shows the item on top of stack
int peek(STACK *Stack);

//Check if stack is empty
int is_empty(STACK *Stack);

int main()
{
    STACK *Stack = create_stack();//We must create an pointer to storage the address of stack
    push(Stack, 10);
    push(Stack, 20);
    push(Stack, 15);
    push(Stack, 40);
    int x, y;
    x = pop(Stack);
    printf("pop: %d\n", x);
    y = peek(Stack);
    printf("peek: %d\n", y);
    x = pop(Stack);
    printf("pop: %d\n", x);
    y = peek(Stack);
    printf("peek: %d\n", y);
    x = pop(Stack);
    printf("pop: %d\n", x);
    y = peek(Stack);
    printf("peek: %d\n", y);
    x = pop(Stack);
    printf("pop: %d\n", x);
    return 0;
}

STACK* create_stack(){
    STACK *new_stack = (STACK*)malloc(sizeof(STACK));//Dynamic allocation
    new_stack->current_size = 0;//Initializing stack size with 0
    return new_stack;//Return the stack pointer
}

void push(STACK *Stack, int item){
    if (Stack->current_size == MAX_STACK_SIZE)//First we must check before add items on stack to see if it isn't full
    {
        printf("Stack overflow\n");
    }
    else
    {
        Stack->items[Stack->current_size++] = item;//The first item of stack is stored in array
    }
}

int pop(STACK *Stack){
    if (is_empty(Stack))//We can't pop from an empty stack
    {
        printf("Stack underflow\n");
        return -1;
    }
    else
    {
        return Stack->items[--Stack->current_size];
    }
}

int peek(STACK *Stack){
    if (is_empty(Stack))//We can't peek from an empty stack
    {
        printf("Stack underflow\n");
        return -1;
    }
    else
    {
        return Stack->items[Stack->current_size - 1];
    }   
}

int is_empty(STACK *Stack){
    if (Stack->current_size == 0)
    {
        return 1;
    }
    return 0;
}

File: Stack/Python/stack.py
#stack implementation using a singly linked list
#Node of list
class Node:
    def __init__(self, value):
        self.value = value
        self.next = None

class Stack:
    #This method initializes the stack
    #Initializing stack
    def __init__(self):
        self.head = Node("head")
        self.size = 0

    #string representation of stack
    def __str__(self):
        current = self.head.next
        out = ""
        while current:
           out += str(current.value) + "->"
           current = current.next
        return out[:-3]

    #get size of stack
    def getSize(self):
        return self.size

    #check if stack is empty
    def isEmpty(self):
        return self.size == 0

    #Get the item on top of stack
    def peek(self):
        # Checking if stack is empty before getting the top item
        if self.isEmpty():
            raise Exception("Stack is empty!")
        return self.head.next.value

    #Insert a value into the stack
    def push(self, data):
        new_node = Node(data)
        new_node.next = self.head.next
        self.head.next = new_node
        self.size += 1

    #Remove from stack and return
    def pop(self):
        if self.isEmpty():
            raise Exception("Stack is empty!")
        remove = self.head.next
        self.head.next = self.head.next.next
        self.size -= 1
        return remove.value

#Start program
if __name__ == "__main__":
    stack = Stack()
    for i in range(1,11):
        stack.push(i)
    print(f"Stack: {stack}")

    for _ in range(1,6):
        remove = stack.pop()
        print(f"Pop: {remove}")
    print(f"Stack: {stack}")